test/test_build_python.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2019 Rover Robotics
+# Licensed under the Apache License, Version 2.0
+import asyncio
+from pathlib import Path
+from types import SimpleNamespace
+
+import colcon_core.shell
+from colcon_core.package_descriptor import PackageDescriptor
+from colcon_core.plugin_system import SkipExtensionException
+from colcon_core.shell.bat import BatShell
+from colcon_core.shell.sh import ShShell
+from colcon_core.subprocess import new_event_loop
+from colcon_core.task import TaskContext
+from colcon_core.task.python.build import PythonBuildTask
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def monkey_patch_get_shell_extensions(monkeypatch):
+    a_shell = None
+    for sh in [ShShell, BatShell]:
+        try:
+            a_shell = sh()
+        except SkipExtensionException:
+            pass
+
+    if a_shell is None:
+        pytest.skip()
```

Comment:
I don't see when this would ever be the case. If we would introduce a new platform with a new primary shell (not very likely) it should be added to this test rather than skipped.